lib/systems/parse.nix

```diff
@@ -166,7 +197,41 @@ rec {
   # Note: Since 22.11 the archs of a mode switching CPU are no longer considered
   # pairwise compatible. Mode switching implies that binaries built for A
   # and B respectively can't be executed at the same time.
-  isCompatible = a: b: with cpuTypes; lib.any lib.id [
+  isCompatible = let
```

Comment:
This was a sensible use of `with`, and duplicating every occurrence of CPU types in this block is exclusively detrimental to maintainability IMO, especially now that you made all other symbols explicit.

Reply:
I'll revert. Sorry for the failure to apply feedback here.